src/rest_vol_dataset.c

```diff
@@ -982,6 +988,25 @@ RV_dataset_write(size_t count, void *dset[], hid_t mem_type_id[], hid_t _mem_spa
                 FUNC_GOTO_ERROR(H5E_DATATYPE, H5E_BADVALUE, FAIL, "memory datatype is invalid");
 
             write_body_len = (size_t)file_select_npoints * dtype_size;
+            if ((contiguous = RV_dataspace_selection_is_contiguous(transfer_info[i].mem_space_id)) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_BADVALUE, FAIL,
+                                "Unable to determine if the dataspace selection is contiguous");
+            if (!contiguous) {
+                if (NULL == (transfer_info[i].u.write_info.write_body = (char *)RV_malloc(write_body_len)))
+                    FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                    "can't allocate space for the 'write_body' values");
+                if (H5Dgather(transfer_info[i].mem_space_id, buf[i], transfer_info[i].mem_type_id,
+                              write_body_len, transfer_info[i].u.write_info.write_body, NULL,
+                              transfer_info[i].u.write_info.write_body) < 0)
```

Comment:
Not an issue really, but you can probably pass NULL as the last parameter here since NULL was provided for H5Dgather's callback function.

Reply:
Ah that is much better. I have implemented the change.